addons/elasticsearch/elasticsearch.yaml

```diff
@@ -0,0 +1,101 @@
+---
+apiVersion: kubeaddons.mesosphere.io/v1beta1
+kind: Addon
+metadata:
+  name: elasticsearch/enterprise
```

Comment:
The name should be `elasticsearch` only. I think we should use namespace other than `kubeaddons` (which is default, I guess) for the catalog services. A separate namespace would allow us to have all the services without any conflict with the cluster services.
WDYT @kaiwalyajoshi and @rishabh96b ?

Reply:
Thanks, Farhan! But but existing addons in kubeaddons-enterprise repo do not use namespace currently (use default) in metadata section, so if we introduce separate namespace we seem have to update all existing addons accordingly. That's the point. Also I've checked KBA-s and have found that 7 out of 33 addons also do not specify namespace explicitly. However, the remaining addons use not only 'kubeaddon' namespace but 'velero', 'istio-system', 'cert-manager' in their Spec section. Having this fact, I think that if we are going to introduce new namespace like 'kubeaddons-enterprise' we DON'T need to move all existing enterprise addons to new namespace but only that ones that can suffer from conflicts with cluster services. I think this approach is quite reasonable and so we don't have to modify the existing addons.

Reply:
Hey guys,

2 comments:

1. the name `elasticsearch/enterprise` is syntactically incorrect, please see https://kubernetes.io/docs/concepts/overview/working-with-objects/names/ for further details.
1. Introducing a custom namespace for enterprise Addons sounds like a good way forward to keep them visibly separate from KBA and other Addons (like Kommander).